Sherpa fails to compile on CI because conda install now uses gcc15

Sherpa fails to build with current main because of an error in `external/regions` as seen in the CI for #2302 and #2302, e.g. see https://github.com/sherpa/sherpa/actions/runs/15123682550/job/42579736969?pr=2303 for full output. Because the build logs are not saved indefinitely, I'll also cope and paste the relevant parts below:
```
  libtool: compile:  /home/runner/work/sherpa/sherpa/conda_loc/envs/build/bin/x86_64-conda-linux-gnu-cc -DPACKAGE_NAME=\"region\" -DPACKAGE_TARNAME=\"region\" -DPACKAGE_VERSION=\"4.14\" "-DPACKAGE_STRING=\"region 4.14\"" -DPACKAGE_BUGREPORT=\"cxchelp@head.cfa.harvard.edu\" -DPACKAGE_URL=\"\" -DPACKAGE=\"region\" -DVERSION=\"4.14\" -DYYTEXT_POINTER=1 -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DSTDC_HEADERS=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_LIBM=1 -DHAVE_FLOAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE__BOOL=1 -DHAVE_MALLOC=1 -DHAVE_REALLOC=1 -DHAVE_SQRT=1 -DHAVE_STRCASECMP=1 -DHAVE_STRCHR=1 -DHAVE_STRPBRK=1 -I. -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /home/runner/work/sherpa/sherpa/conda_loc/envs/build/include -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /home/runner/work/sherpa/sherpa/conda_loc/envs/build/include -MT region_extent.lo -MD -MP -MF .deps/region_extent.Tpo -c region_extent.c  -fPIC -DPIC -o region_extent.o
  region_eval.c: In function 'regCompareRegion':
  region_eval.c:83:9: error: expected identifier or '(' before 'true'
     83 |     int true  = 1;
        |         ^~~~
  region_eval.c:84:9: error: expected identifier or '(' before 'false'
     84 |     int false = 0;
        |         ^~~~~
  make[3]: *** [Makefile:514: region_eval.lo] Error 1
  make[3]: *** Waiting for unfinished jobs....
  mv -f .deps/region_extent.Tpo .deps/region_extent.Plo
  libtool: compile:  /home/runner/work/sherpa/sherpa/conda_loc/envs/build/bin/x86_64-conda-linux-gnu-cc -DPACKAGE_NAME=\"region\" -DPACKAGE_TARNAME=\"region\" -DPACKAGE_VERSION=\"4.14\" "-DPACKAGE_STRING=\"region 4.14\"" -DPACKAGE_BUGREPORT=\"cxchelp@head.cfa.harvard.edu\" -DPACKAGE_URL=\"\" -DPACKAGE=\"region\" -DVERSION=\"4.14\" -DYYTEXT_POINTER=1 -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DSTDC_HEADERS=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_LIBM=1 -DHAVE_FLOAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE__BOOL=1 -DHAVE_MALLOC=1 -DHAVE_REALLOC=1 -DHAVE_SQRT=1 -DHAVE_STRCASECMP=1 -DHAVE_STRCHR=1 -DHAVE_STRPBRK=1 -I. -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /home/runner/work/sherpa/sherpa/conda_loc/envs/build/include -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /home/runner/work/sherpa/sherpa/conda_loc/envs/build/include -MT region_logic.lo -MD -MP -MF .deps/region_logic.Tpo -c region_logic.c  -fPIC -DPIC -o region_logic.o
  libtool: compile:  /home/runner/work/sherpa/sherpa/conda_loc/envs/build/bin/x86_64-conda-linux-gnu-cc -DPACKAGE_NAME=\"region\" -DPACKAGE_TARNAME=\"region\" -DPACKAGE_VERSION=\"4.14\" "-DPACKAGE_STRING=\"region 4.14\"" -DPACKAGE_BUGREPORT=\"cxchelp@head.cfa.harvard.edu\" -DPACKAGE_URL=\"\" -DPACKAGE=\"region\" -DVERSION=\"4.14\" -DYYTEXT_POINTER=1 -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DSTDC_HEADERS=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_LIBM=1 -DHAVE_FLOAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE__BOOL=1 -DHAVE_MALLOC=1 -DHAVE_REALLOC=1 -DHAVE_SQRT=1 -DHAVE_STRCASECMP=1 -DHAVE_STRCHR=1 -DHAVE_STRPBRK=1 -I. -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /home/runner/work/sherpa/sherpa/conda_loc/envs/build/include -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /home/runner/work/sherpa/sherpa/conda_loc/envs/build/include -MT region_repr.lo -MD -MP -MF .deps/region_repr.Tpo -c region_repr.c  -fPIC -DPIC -o region_repr.o
  libtool: compile:  /home/runner/work/sherpa/sherpa/conda_loc/envs/build/bin/x86_64-conda-linux-gnu-cc -DPACKAGE_NAME=\"region\" -DPACKAGE_TARNAME=\"region\" -DPACKAGE_VERSION=\"4.14\" "-DPACKAGE_STRING=\"region 4.14\"" -DPACKAGE_BUGREPORT=\"cxchelp@head.cfa.harvard.edu\" -DPACKAGE_URL=\"\" -DPACKAGE=\"region\" -DVERSION=\"4.14\" -DYYTEXT_POINTER=1 -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DSTDC_HEADERS=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_LIBM=1 -DHAVE_FLOAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE__BOOL=1 -DHAVE_MALLOC=1 -DHAVE_REALLOC=1 -DHAVE_SQRT=1 -DHAVE_STRCASECMP=1 -DHAVE_STRCHR=1 -DHAVE_STRPBRK=1 -I. -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /home/runner/work/sherpa/sherpa/conda_loc/envs/build/include -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /home/runner/work/sherpa/sherpa/conda_loc/envs/build/include -MT region_shape.lo -MD -MP -MF .deps/region_shape.Tpo -c region_shape.c  -fPIC -DPIC -o region_shape.o
  mv -f .deps/region_logic.Tpo .deps/region_logic.Plo
  mv -f .deps/region_repr.Tpo .deps/region_repr.Plo
  mv -f .deps/region_shape.Tpo .deps/region_shape.Plo
  make[3]: Leaving directory '/home/runner/work/sherpa/sherpa/extern/region-4.14/src'
  make[2]: *** [Makefile:646: install] Error 2
  make[2]: Leaving directory '/home/runner/work/sherpa/sherpa/extern/region-4.14/src'
  make[1]: *** [Makefile:385: install-recursive] Error 1
  make[1]: Leaving directory '/home/runner/work/sherpa/sherpa/extern/region-4.14'
  make: *** [Makefile:386: install-recursive] Error 1
  error: subprocess-exited-with-error
  
  × Building editable for sherpa (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /home/runner/work/sherpa/sherpa/conda_loc/envs/build/bin/python3.10 /home/runner/work/sherpa/sherpa/conda_loc/envs/build/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py build_editable /tmp/tmp06h5s46i
  cwd: /home/runner/work/sherpa/sherpa
  Building editable for sherpa (pyproject.toml): finished with status 'error'
  ERROR: Failed building editable for sherpa
Failed to build sherpa
ERROR: Failed to build installable wheels for some pyproject.toml based projects (sherpa)
Error: Process completed with exit code 1.
```

Why does the rero surface now, but didn't in rc testing, I don't know. I can only suspect that it's something to do with the C++ version and the dependencies that are on the current GH runs - but it could be something else...

--- a/.github/scripts/setup_conda.sh
+++ b/.github/scripts/setup_conda.sh
@@ -22,7 +22,7 @@ if [ "`uname -s`" == "Darwin" ] ; then
     tar -C ${GITHUB_WORKSPACE}/11.0SDK -xf MacOSX11.0.sdk.tar.xz
 
 else
-    compilers="gcc_linux-64 gxx_linux-64 gfortran_linux-64"
+    compilers="gcc_linux-64=14.2 gxx_linux-64=14.2 gfortran_linux-64"
 
     if [ -n "${MATPLOTLIBVER}" ]; then
         #Installed for qt-main deps which is needed for the QtAgg backend to work for matplotlib

--- a/conda_build_config.yaml
+++ b/conda_build_config.yaml
@@ -1,2 +1,8 @@
 CONDA_BUILD_SYSROOT:
   - /opt/MacOSX11.0.sdk        # [osx]
+
+c_compiler_version:    # [linux]
+    - 14.2             # [linux]
+cxx_compiler_version:  # [linux]
+    - 14.2             # [linux]
+

